internal/sftp/goclient_test.go

```diff
@@ -288,3 +287,75 @@ func TestGoClient(t *testing.T) {
 		})
 	}
 }
+
+func DeleteTest(t *testing.T) {
+	host, port, err := net.SplitHostPort(serverAddress)
+	if err != nil {
+		t.Fatalf("Bad server address: %s", serverAddress)
+	}
+
+	_ = startSFTPServer(t, serverAddress)
+
+	// Start a listener on an open port and use the address to test a bad SFTP
+	// server address.
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatalf("Couldn't start listener: %v", err)
+	}
+	defer listener.Close()
+
+	type test struct {
+		name     string
+		cfg      sftp.Config
+		bytes    int64
+		filename string
+		wantErr  string
+	}
+	for _, tc := range []test{
+		{
+			name: "Deletes a file",
+			cfg: sftp.Config{
+				Host:           host,
+				Port:           port,
+				KnownHostsFile: "./testdata/known_hosts",
+				PrivateKey: sftp.PrivateKey{
+					Path: "./testdata/clientkeys/test_ed25519",
+				},
+			},
+			bytes:    13,
+			filename: "test.txt",
+		},
+		{
+			name: "Errors when file doesn't exist",
+			cfg: sftp.Config{
+				Host:           host,
+				Port:           port,
+				KnownHostsFile: "./testdata/known_hosts",
+				PrivateKey: sftp.PrivateKey{
+					Path: "./testdata/clientkeys/test_ed25519",
+				},
+			},
+			filename: "error.txt",
+			wantErr:  "SFTP: unable to get file info for",
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			sftpc := sftp.NewGoClient(tc.cfg)
+			dest := tfs.NewDir(t, "sftp_test")
+			pre := dest.Join("test.txt")
```

Comment:
This can be combined with the `tfs.NewDir()` method:

```
dest := tfs.NewDir(t, "sftp_test", tfs.WithFile("test.txt", ""))
```

Reply:
it can be but I think the way it is clearer

Reply:
actually nope i agree with you now. just had to see it.